src/os_win/os_errno.c

```diff
@@ -63,14 +63,33 @@ int
 __wt_errno(void)
 {
 	/*
+	 * Check for 0:
+	 * It's easy to introduce a problem by calling the wrong error function,
+	 * for example, this function when the MSVC function set the C runtime
+	 * error value. Handle gracefully and always return an error.
+	 */
+	return (errno == 0 ? WT_ERROR : errno);
```

Comment:
@agorrod 
> This change to Windows error handling bothers me.

I think you're missing part of this change. In the Windows code, after a POSIX or ANSI C error, we call `__wt_errno()`, after a Windows error we call `__wt_win32_errno()`, so it looks like this:

```
if (filehandle_secondary == INVALID_HANDLE_VALUE)
    WT_RET_MSG(session,
        __wt_win32_errno(), "%s: handle-open: CreateFileA: secondary", name);

if (fclose(fh->fp) != 0)  
    WT_RET_MSG(session,
        __wt_errno(), "%s: handle-close: fclose", fh->name);
```

There's an additional change, to always return an error value, and that's because it's easy to make a mistake and call the wrong error function (the Windows documentation isn't clear in some cases, too).

If the wrong error function gets called, we could return "success" out of the middle of a function when we should have returned an error. And, of course, these are error paths, so we're less likely to test them before a release, and if a call fails in the field we could be surprised.

By always returning an error code out of those functions, we ensure correct behavior in the caller.